src/main/java/uk/gov/hmcts/dts/fact/controllers/admin/AdminCourtAddressController.java

```diff
@@ -68,25 +68,35 @@ public ResponseEntity<List<CourtAddress>> getCourtAddresses(@PathVariable String
      * Replace the existing addresses with the new ones for a specific court.
      * @param slug Court slug
      * @param courtAddresses a list of court addresses to be updated
-     * @return A list of updated court addresses
-     *         If one of more input address postcodes are invalid, return the invalid postcodes and a '400' response.
+     * @return A list of updated court addresses.
+     *     If one of more input address postcodes or epim ids are invalid, return them with a '400' response.
      */
     @PutMapping(path = "/{slug}/addresses")
     @Operation(summary = "Update addresses for a provided court")
     @ApiResponse(responseCode = "200", description = "Successful")
-    @ApiResponse(responseCode = "400", description = "Invalid postcodes")
+    @ApiResponse(responseCode = "400", description = "Invalid postcodes or epim IDs")
     @ApiResponse(responseCode = "401", description = "Unauthorized")
     @ApiResponse(responseCode = "403", description = "Forbidden")
     @ApiResponse(responseCode = "404", description = "Court not Found")
     @Role({FACT_ADMIN, FACT_SUPER_ADMIN})
     public ResponseEntity<List<CourtAddress>> updateCourtAddresses(@PathVariable String slug,
                                                                    @RequestBody List<CourtAddress> courtAddresses,
                                                                    Authentication authentication) {
-        final List<String> invalidPostcodes = adminService.validateCourtAddressPostcodes(courtAddresses);
-        if (CollectionUtils.isEmpty(invalidPostcodes)) {
-            adminCourtLockService.updateCourtLock(slug, authentication.getName());
-            return ok(adminService.updateCourtAddressesAndCoordinates(slug, courtAddresses));
+
+        if (!CollectionUtils.isEmpty(courtAddresses)) {
```

Comment:
This can be moved into the service. In the controller it should pretty much call the service method responsible.

Reply:
changed